core/scheduler/service/job_sla_predictor_service.go

```diff
@@ -0,0 +1,294 @@
+package service
+
+import (
+	"context"
+	"time"
+
+	"github.com/goto/optimus/core/scheduler"
+	"github.com/goto/optimus/core/tenant"
+	"github.com/goto/salt/log"
+)
+
+type DurationEstimator interface {
+	GetP95DurationByJobNames(ctx context.Context, jobNames []scheduler.JobName, lastNRuns int, bufferPercentage int) (map[scheduler.JobName]*time.Duration, error)
+}
+
+type JobState struct {
+	JobSLAState
+	JobName       scheduler.JobName
+	Tenant        tenant.Tenant
+	RelativeLevel int
+	Status        string
+}
+
+type JobSLAState struct {
+	EstimatedDuration *time.Duration
+	InferredSLA       *time.Time
+}
+
+type JobSLAPredictorService struct {
+	l                 log.Logger
+	lastNRuns         int
+	jobLineageFetcher JobLineageFetcher
+	durationEstimator DurationEstimator
+}
+
+func NewJobSLAPredictorService(l log.Logger, jobLineageFetcher JobLineageFetcher, durationEstimator DurationEstimator) *JobSLAPredictorService {
+	return &JobSLAPredictorService{
+		l:                 l,
+		lastNRuns:         7,
```

Comment:
lastNruns & buffer percentage can be configuration in Optimus & should be state variables in duration estimator